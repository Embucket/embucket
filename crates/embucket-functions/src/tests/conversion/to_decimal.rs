use crate::test_query;

test_query!(
    to_decimal_basic,
    "SELECT column1,
       TO_NUMBER(column1),
       TO_NUMBER(column1, 10, 1),
       TO_NUMBER(column1, 10, 8)
    FROM VALUES ('12.3456'), ('98.76546')",
    snapshot_path = "to_decimal"
);

test_query!(
    to_decimal_null,
    "SELECT column1, TRY_TO_NUMBER(column1, 10, 9)
    FROM VALUES ('12.3456'), ('98.76546')",
    snapshot_path = "to_decimal"
);

test_query!(
    to_decimal_format,
    "SELECT column1,
       TO_DECIMAL(column1, '99.9') as D0,
       TO_DECIMAL(column1, '99.9', 9, 5) as D5,
       TO_DECIMAL(column1, 'TM9', 9, 5) as TD5
    FROM VALUES ('1.0'), ('-12.3'), ('0.0'), ('- 0.1');",
    snapshot_path = "to_decimal"
);

test_query!(
    to_decimal_format_comma,
    "SELECT column1,
       TO_DECIMAL(column1, '9,999.99', 6, 2) as convert_number
    FROM VALUES ('3,741.72')",
    snapshot_path = "to_decimal"
);

test_query!(
    to_decimal_format_dollar_sign,
    "SELECT column1,
       TO_DECIMAL(column1, '$9,999.99', 6, 2) as convert_currency
    FROM VALUES ('$3,741.72')",
    snapshot_path = "to_decimal"
);
