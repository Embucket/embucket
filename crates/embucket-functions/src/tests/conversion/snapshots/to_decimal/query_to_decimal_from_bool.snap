---
source: crates/embucket-functions/src/tests/conversion/to_decimal.rs
description: "\"SELECT column1,\n       TO_DECIMAL(column1) as convert_bool\n    FROM VALUES (FALSE), (TRUE)\""
---
Ok(
    [
        "+---------+--------------+",
        "| column1 | convert_bool |",
        "+---------+--------------+",
        "| false   | 0            |",
        "| true    | 1            |",
        "+---------+--------------+",
    ],
)
